onnxruntime/core/providers/cpu/ml/tree_ensemble_classifier.cc

```diff
@@ -173,7 +71,21 @@ common::Status TreeEnsembleClassifier<T>::Compute(OpKernelContext* context) cons
   Tensor* Y = context->Output(0, {N});
   Tensor* Z = context->Output(1, {N, tree_ensemble_.get_class_count()});
 
-  tree_ensemble_.compute(context, &X, Z, Y);
+  ORT_IF_CONSTEXPR(std::is_same<TH, TO>::value) {
+    TH* z_data = Z->template MutableData<TH>();
+    tree_ensemble_.compute(context, &X, z_data, Y);
+  } else {
+    int64_t size = Z->Shape().Size();
+    std::unique_ptr<TH> z_data_th(new TH[size]);
+    TH* tree_output = z_data_th.get();
+    tree_ensemble_.compute(context, &X, tree_output, Y);
+    TO* z_data = Z->template MutableData<TO>();
+    for (int64_t i = 0; i < size; ++i) {
+      z_data[i] = static_cast<TO>(tree_output[i]);
+    }
+    z_data_th.release();
```

Comment:
>z_data_th.release(); [](http://example.com/codeflow?start=4&length=20)

This will leak memory

Reply:
fixed, I should have added WIP for this iteration.